### ISSUE TITLE
Problems with OSX folder permissions not properly reported

### ISSUE THREAD

**Issue by peterhgruber** [microsoft/vscode#207722]
Type: <b>Bug</b>

(1) In OSX System Settings > Privacy & Security > Files and Folders > VS Code disallow acces to a folder, e.g. Documents
(2) Open a file, e.g. a Jupyter notebook from within such a folder in VS Code. This works!
(3) Try to run the file
(4) One obtains a very CRYPTIC error message 
```
Failed to start the Kernel. 
EPERM: operation not permitted, scandir '/Users/xxx/Documents/myproject'. 
View Jupyter log for further details.
````
Conclusion: VS code does not handle OSX folder permissions correctly and does not report properly the problems of accessing e.g. the Desktop or Documents folder. It should ask for permission at the OS level and furthermore present a more coherent error message.

VS Code version: Code 1.87.2 (Universal) (863d2581ecda6849923a2118d93a088b0745d9d6, 2024-03-08T15:21:31.043Z)
OS version: Darwin arm64 22.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Max (12 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|18, 12, 8|
|Memory (System)|32.00GB (1.95GB free)|
|Process Argv|--crash-reporter-id 0746296c-8cee-4a25-8c7c-21937d1460b0|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
codespaces|Git|1.16.13
copilot|Git|1.172.0
copilot-chat|Git|0.13.1
marp-vscode|mar|2.8.0
language-matlab|Mat|1.2.0
debugpy|ms-|2024.2.0
python|ms-|2024.2.1
vscode-pylance|ms-|2024.3.1
jupyter|ms-|2024.2.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.17
vscode-jupyter-cell-tags|ms-|0.1.8
vscode-jupyter-slideshow|ms-|0.1.5
remote-ssh|ms-|0.109.0
remote-ssh-edit|ms-|0.86.0
live-server|ms-|0.4.13
remote-explorer|ms-|0.4.3
r|REd|2.8.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:30979542
d34g3935:30971562
fegfb526:30981948
bg6jg535:30979843
ccp2r3:30958157
dsvsc020:30976470
gee8j676:30988843

```

</details>

**Comment by lszomoru** [comment]
@deepak1556, could you help triage this issue? Thank you!

**Comment by vinay-swamy** [comment]
This is obviously not the best way to do this, but for anyone else who encounters the same error and needs an immediate fix, giving full disk access to vscode can fix this error. In settings go to Privacy > Full Disk Access and add VScode.

**Comment by deepak1556** [comment]
We use fs implementation from Node.js and we don't deal with the OS permissions directly, moving to @bpasero for thoughts on improving error feedback from the fs layer.

**Comment by artdfel** [comment]
Heavy +1. I stumbled upon this issue when setting up jupyter notebook with R kernel.
